pkgs/by-name/la/lammps/package.nix

```diff
@@ -44,9 +45,27 @@
   extraCmakeFlags ? { },
```

Comment:
It's true the semantics of `config.cudaSupport` is somewhat vague. I think my expectation would be something along the lines of `extraCmakeFlags ? optionalAttrs config.cudaSupport { GPU_API = "cuda"; }`, which would parallel how other packages accept `cudaSupport ? config.cudaSupport`

Reply:
Alternatively, one could cut off some of the combinations of arguments using `assert`s...

Reply:
> It's true the semantics of `config.cudaSupport` is somewhat vague. I think my expectation would be something along the lines of `extraCmakeFlags ? optionalAttrs config.cudaSupport { GPU_API = "cuda"; }`, which would parallel how other packages accept `cudaSupport ? config.cudaSupport`

Yes but that won't be enough to make the build change anything. You also have to set `packages.GPU` to `true` (or CMake-wise: `-DPKG_GPU=ON`). See my table outside the review comments.